osu.Game.Rulesets.Osu/UI/Cursor/CursorTrail.cs

```diff
@@ -174,10 +183,10 @@ protected void AddTrail(Vector2 position)
                     float distance = diff.Length;
                     Vector2 direction = diff / distance;
 
-                    float interval = partSize.X / 2.5f * IntervalMultiplier;
-                    float stopAt = distance - (AvoidDrawingNearCursor ? interval : 0);
+                    Vector2 interval = partSize.X / 2.5f * IntervalMultiplier * scaleRatio;
```

Comment:
It sounds like you should be able to remove the scale from `partSize`, and, along with that, use `Texture.DisplayWidth` directly (and remove the two `LayoutValue`s? This is the only usage of it after all.

Have you also made sure that the trail visually looks the same as in current master?

Reply:
> Have you also made sure that the trail visually looks the same as in current master?

yes, that's why I use extract scaleRatio from DrawInfo.Matrix

Reply:
>use `Texture.DisplayWidth` directly (and remove the two `LayoutValue`s? This is the only usage of it after all.

do I have to use a vector? I don't know if the DisplayWidth is guaranteed equal to DisplayHeight.

Reply:
There's three choices:
- `new Vector2(Texture.DisplayWidth, Texture.DisplayHeight)`
- `new Vector2(Texture.DisplayWidth)`
- Or go back to using `float` and just `Texture.DisplayWidth`

Personally, I would choose (3) just so that we don't inadvertently break something else with this PR, but I'm not sure why you've gone to using a vector other than that it looks more correct, which I would be willing to entertain.

Reply:
Actually, if there's a strong reason not to, then I would suggest doing (3) for a "fix" PR.

Reply:
> but I'm not sure why you've gone to using a vector other than that it looks more correct, which I would be willing to entertain.

not for special reasons, just because the extracted scale is a vector and in case some parents scale the X-axis and Y-axis differently.